### PR TITLE
dd-trace-go v1.36.0 and configurable build tags

### PR DESCRIPTION
**Pull request by Julio-Guerra**
- Update go.mod dd-trace-go v1.36.0
- Make v1.36.0 the default dd-trace-go version
- Add a new Makefile var `BUILD_TAGS` that can be externally configured

This way we can update rel-env's `./build.sh` script by calling the makefile this way when appsec is enabled:
```console
$ BUILD_TAGS=appsec make install BUILD_VARIANT="${1}" DD_TRACE_GO_VERSION="${2}" # note the env var vs command arg difference
```

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,8 +1,9 @@
-BUILD_VARIANT=release
-DD_TRACE_GO_VERSION:=v1.34.0
+BUILD_VARIANT       ?= release
+DD_TRACE_GO_VERSION ?= v1.36.0
+BUILD_TAGS          := $(BUILD_VARIANT),$(BUILD_TAGS)
```
Comment (felixge) [inline Makefile:3]: does it make sense to have both `BUILD_VARIANT` and `BUILD_TAGS`? Or wouldn't it be simpler to just have `BUILD_TAGS`?

**Review by felixge — APPROVED** [review]
LGTM – not sure if my comment makes sense, so feel free to ignore it.

File: Makefile (hunk 1 above)
Comment (Julio-Guerra) [reply Makefile:3]: Sure yes. I felt like you wanted "business-specific" parameters so I added a separate BUILD_TAGS one, but all good to merge both into a single one.

Comment (Julio-Guerra) [reply Makefile:3]: My bad @felixge: it's actually not possible due to the following `-X main.version=$(BUILD_VARIANT)/git describe --tags HEAD` that would be impossible to deduce out of `BUILD_TAGS` alone.

WDYT? Do I merge the way it is?

Comment (felixge) [reply Makefile:3]: Yeah, let's just merge it. We can change it again anytime.

**Pull request merged**